src/stats/lexd.rs

```diff
@@ -0,0 +1,57 @@
+use std::collections::HashSet;
+
+use tree_sitter::{Language, Parser, TreeCursor};
+
+use rocket_contrib::{json, json::JsonValue};
+
+use slog::Logger;
+
+use crate::{models::StatKind, stats::StatsError};
+
+extern "C" {
+    fn tree_sitter_lexd() -> Language;
+}
+
+pub fn get_stats(_logger: &Logger, body: &str) -> Result<Vec<(StatKind, JsonValue)>, StatsError> {
+    let mut parser = Parser::new();
```

Comment:
If you wanted to have even more fun, you could wrap this in an object pool: https://github.com/zslayton/lifeguard and `reset` it for new parses.

Reply:
In principle the parser object could be created at start time and just kept around (that's what I do in apertium-lint, though that's not set up to be kept running).